src/Http/Client.php

```diff
@@ -181,25 +201,25 @@ private function parseResponse(ResponseInterface $response)
         }
 
         if ($response->getStatusCode() >= 300) {
-            $body = $this->json->unserialize((string) $response->getBody()) ?? $response->getReasonPhrase();
+            $body = $this->json()->unserialize((string) $response->getBody()) ?? $response->getReasonPhrase();
 
             throw new ApiException($response, $body);
         }
 
-        return $this->json->unserialize((string) $response->getBody());
+        return $this->json()->unserialize((string) $response->getBody());
     }
 
     /**
      * Checks if any of the header values indicate a JSON response.
      *
      * @param array $headerValues the array of header values to check
      *
-     * @return bool true if any header value contains 'application/json', otherwise false
+     * @return bool true if any header value contains 'application/json' or 'application/x-ndjson', otherwise false
      */
     private function isJSONResponse(array $headerValues): bool
     {
         $filteredHeaders = array_filter($headerValues, static function (string $headerValue) {
-            return false !== strpos($headerValue, 'application/json');
+            return str_contains($headerValue, 'application/json') || str_contains($headerValue, 'application/x-ndjson');
```

Comment:
str_contains is available only from php 8, this library supports php 7.4, so you cannot use it yet